src/main/scala/org/apache/spark/sql/delta/commands/WriteIntoDelta.scala

```diff
@@ -101,7 +101,8 @@ case class WriteIntoDelta(
       deltaLog.fs.mkdirs(deltaLog.logPath)
     }
 
-    val newFiles = txn.writeFiles(data, Some(options))
+    val canChangeData = options.canChangeData
+    val newFiles = txn.writeFiles(data, Some(options)).map(_.copy(dataChange = canChangeData))
```

Comment:
do we have to always copy in the case where dataChange is true ?

Reply:
no, but even when copy, it will be copied to it's actual value which true

Reply:
Yeah then we can have an if condition to copy only when canChangeData is false, because its always true by default, we can avoid the map and copy operation

Reply:
ok, I will do it